src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -29,30 +29,50 @@ public async Task TestEmptyFile()
             await VerifyCS.VerifyRefactoringAsync(code, code);
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
-        public async Task TestSimpleReferenceType()
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        [InlineData("", true)]
+        [InlineData("csharp_style_prefer_parameter_null_checking = true", true)]
+        [InlineData("csharp_style_prefer_parameter_null_checking = false", false)]
+        public async Task TestSimpleReferenceType(string editorConfig, bool useParameterNullChecking)
```

Comment:
💡 Consider rewriting this to minimize unnecessary/duplicate parameterization of the method:

```suggestion
        [InlineData(null)]
        [InlineData(true)]
        [InlineData(false)]
        public async Task TestSimpleReferenceType(bool? useParameterNullChecking)
```

Reply:
My preference would be to extract the scenario where the option is explicitly disabled to its own `[Fact]`, and then treat this test as though `useParameterNullChecking` is a constant true. However, I don't feel strongly about any particular factoring of the test.

Reply:
@sharwell @RikkiGibson I'm a bit confused as to which suggestion should I take. Both sounds reasonable and I'm okay either way.

Reply:
Let's go with https://github.com/dotnet/roslyn/pull/59561#discussion_r807190446 :)

Reply:
The downside of the approach mentioned by @RikkiGibson is we lose the ability to see that the two tests are operating on different input. At that point, it's not clear whether the different output is caused by a change in the option, or a change in the input. While this is rarely a problem in the pull request where the test is introduced, it's easy and common for an update to one test to fail to realize that the other is a matching test that needs to be updated at the same time. For this reason, it is strongly recommended for the test to be combined.

Reply:
@Youssef1313 I won't block this PR on the suggested change. More a note for the future to keep with the approach you were already using 😄